### DISCUSSION TITLE
Nw one with explorers

### DISCUSSION THREAD

**Discussion by StaSiii** [fedorovvl/tso_client#242] [Ideas]
- it would be nice to have such an option, especially at events, that simply if I'm not there for 8 hours, the explorers won't be sitting in the star

I'll start with the game mechanics at events, because that's why such a special functionality is needed.
For all events except New Year's, the scaling of bringing event items is against players. This means that the longer explorers are sent to search, the less they bring.
Example on the upcoming Easter event:

Short Treasure Search              2             3h
Medium Treasure Search                3               6h
Long Treasure Search              5            12h
Very Long Treasure Search               7            24h
Prolonged Treasure Search            9            36h

If it were properly time scaled, it should look like this:

Short Treasure Search          2          3h
Medium Treasure Search        4            6h
Long Treasure Search         8           12h
Very Long Treasure Search              16         24h
Prolonged Treasure Search              24        36h

The shorter the search, the more eventful items the explorers will bring.

Therefore, my idea for an addition to the client.
The idea is that you could set a time, say 8 hours, and the script would mark all of the displayed explorers within that time. In addition, the script would select different searches for them, those that fit within the given time.
I know it can be done now using templates, but I have 12 of these templates. And it looks like this:
explorers_1h.txt
explorers_2h.txt
explorers_3h.txt
......
explorers_36h.txt

After each event in the game, all these 12 templates need to be corrected, because the player will buy new explorers that need to be added to each of the 12 files. And it's not that easy anymore.

In short:
1. We set the time
2. Click on the special button
3. The script recalculates the current list of explorers and selects those who fall within the given time, setting them different types of searches, but the closest possible in terms of time.
4. The script doesn't mark those who didn't fit in the given time (they may be in red or something).
5. The player's reaction is always needed afterwards, whether to send it as the script established or to correct something.

**Comment by Mazimus332** [comment]
Just in case I misunderstood your idea: You want some new script to import old explorer template with some "search time limit" option to set time limit, depending on that time, explorers will be sent to searches with maximum time but lower than the set time, then export template and use it. You also want that even explorers who aren't available (bc already on search) appear on that list.

It should be kept in mind that Humble Explorer won't do good searches when set to higher than Medium Treasure Search, and Romantic won't do good searches when set to lower than Long Treasure Search.

But I have something that could help you and maybe not need a whole new script for this...

For Easter Event (and almost every event) the best search is Medium for all cases except Romantic Explorer.

![Screenshot_216](https://user-images.githubusercontent.com/116684854/226999393-93afdedb-b2cd-4979-a4f5-86a7f91d6085.png)

You'll need all your explorers with skills to reduce their time, obviously.

![Screenshot_217](https://user-images.githubusercontent.com/116684854/227001804-25a691b9-f511-4823-bc30-61ab25e52894.png)

I recommend you using the following page to calculate better explorer searches for each event.

https://dsowiki.de/wiki/search/Osterevent/2022

Knowing this, I only use 2 different templates for my explorers during events, not 12 as you do, so it saves a lot of time, so keep that in mind.
Template 1: Medium search
Template 2: Long/Very Long/Prolonged search (for nighttime)

I hope this information helps you to save time from re-editing templates for each event, without needing to create a new whole script for this... Even though I think it's a good idea, I think that SirriS won't do it because there should be more important things to do, explorer script already saves a lot of time and works perfectly, obviously several QoL new features can be done but he would prefer to do the most important ones.

**Comment by StaSiii** [comment]
That whole text was translated by google

anyway, there is not ab saving templates
mine templates was as an eample...............
just a button which will find and set explorers when ill come back from work...
saying ab humble and romantic, u can always ucheck them
